tools/api/clone.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+# Simple script to clone vM to vN API, performing sed-style heuristic fixup of
+# build paths and package references.
+#
+# Usage:
+#
+# ./tools/api/clone.sh v2 v3
+
+set -e
+
+declare -r OLD_VERSION="$1"
+declare -r NEW_VERSION="$2"
+
+function replace_build() {
+  sed -i -e "s#\(//$1[^\S]*\)/${OLD_VERSION}alpha#\1/${NEW_VERSION}#g" "$2"
+  sed -i -e "s#\(//$1[^\S]*\)/${OLD_VERSION}#\1/${NEW_VERSION}#g" "$2"
+}
+
+function replace_proto() {
+  sed -i -e "s#\($1\S*[\./]\)${OLD_VERSION}alpha#\1${NEW_VERSION}#g" "$2"
+  sed -i -e "s#\($1\S*[\./]\)${OLD_VERSION}#\1${NEW_VERSION}#g" "$2"
+}
+
+for p in $(find api/ -name "${OLD_VERSION}" -o -name "${OLD_VERSION}alpha")
```

Comment:
Does this cover alpha1? I think we have some of that but not sure. If it does cover it why do we need the explicit alpha or branch?

Reply:
I can't see any `{v1,v1alpha,alphav1,alpha1,alpha2}`. We do the `-o` to allow us to deal with the situation where there is both `{v2alpha,v2}` for a given package. I'll add comments.